doc/src/EntitySpecMeta.dox

```diff
@@ -15,177 +15,309 @@
+ * references" API methods, and cover the description of relationships
+ * not defined by the _type_ of entities themselves.
+ *
+ * An example use case of this would be an editorial application, that
+ * wishes to define a new shot in a production tracking system. Given
+ * the @ref entity_reference of an existing shot on the timeline, a
+ * specification that composed the `relationship` and `partent`
+ * traits allows the host to query the correct parent to register the
+ * new shot to. It explicitly avoids the host having to know what type
+ * of thing the parent is - which would require an entity specification.
+ *
+ * This ensures that this action can work with any asset management
+ * system, without needing semantic understanding of the specific and
+ * most likely propriatory structure maintained by the manager.
+ *
+ * Another example of this would be a specification composing the
+ * `relationship` and `dependency` traits to allow the runtime
+ * dependencies of a specific entity to be queried to facilitate data
+ * relocation.
+ *
+ * @section specification_specificity A Note on Trait Specificity and Entities
+ *
+ * @subsection Using Specifcations as a Filter Predicate
+ *
+ * When specifcations (or @ref trait_collection "trait collections") are
+ * used as a predicate to search for entities. Such as in
+ * @ref openassetio.hostAPI.Manager.Manager.getRelatedReferences
+ * "getRelatedReferences", or when browsing for assets, traits are
+ * considered additive, and they must all be satisifed.
+ *
+ * As an example, consider a world-building tool wishing to browse for
+ * an arbitrary three-dimensional scene file to incorporate (e.g. USD).
+ * If it has no specific interest in exactly what type of objects it
+ * may contain - just that is scene data, it should not include any
+ * traits that cover the specific types of scene items in the
+ * specification.
+ *
+ * It may be tempting to form one composed of all potential traits, eg:
+ * `blob`, `scene`, `camera`, `lights`, `geometry` etc. Because of the
+ * additive nature of a @ref trait_collection, this would effectively
+ * require that _only_ scenes conforming to all of the listed traits
+ * should be presented. The correct trait collection to use would be
+ * just `blob` and `scene`.
+ *
+ * If it was required that the data be in a specific format (eg: USD)
+ * then the `mimeType` property defined by the `blob` trait should be
+ * set accordingly.
+ *
+ * If a host _did_ however require that a scene file contained a certain
+ * type of data  - a camera for example a camera loader, then the
+ * `camera` trait should be added to indicate that only scenes
+ * containing one should be presented.
+ *
+ * Another example would be the when quering for an an EDL file in
+ * relation to a sequence entity. The traits used should only include a
+ * description of the nature of any of its content if this is
+ * specifically required for its use, so typically `blob`, `EDL` is all
```

Comment:
I think the thought was that it would be `lowerCamelCase` 🐫 , to match our other conventions. But as they're opaque strings (and I think inherently UTF-8?) then in theory there are no actual constraints on the format?

Reply:
OK that's fine, I think I just wanted to make sure.